KomaMRIBase/src/datatypes/Phantom.jl

```diff
@@ -304,26 +329,55 @@ Creates a three-dimentional brain Phantom struct.
 ```julia-repl
 julia> obj = brain_phantom3D(; ss=5)
 
+julia> obj = brain_phantom3D(; us=[2, 2, 1])
+
 julia> plot_phantom_map(obj, :ρ)
 ```
 """
-function brain_phantom3D(;ss=4,us=1,start_end=[160, 200])
+function brain_phantom3D(;ss=4, us=1, start_end=[160, 200])
+
+    # check for valid input    
+    @assert length( ss) <= 3 "ss=$(ss) invalid, ss can have up to three components [ssx, ssy, ssz] for a 3D phantom"
+    @assert length( us) <= 3 "us=$(us) invalid, us can have up to three components [usx, usy, usz] for a 3D phantom"
+    if length( us) > 1 || prod( us) > 1
+        @info "setting ss=1 since us=$(us) defined"
+        ss = 1
+    end
+    if length( us) == 1
+        usx = us[1]; usy = us[1]; usz = us[1]
+    elseif length( us) == 2
+        usx = us[1]; usy = us[2]; usz = us[2]
+        @warn "Using us=$([usx, usy, usz]) in place of us=$([usx, usy])."
+    else
+        usx = us[1]; usy = us[2]; usz = us[3]
+    end
+    if length( ss) == 1
+        ssx = ss[1]; ssy = ss[1]; ssz = ss[1]   
+    elseif length( ss) == 2
+        ssx = ss[1]; ssy = ss[2]; ssz = ss[2]
+        @warn "Using ss=$([ssx, ssy, ssz]) in place of ss=$([ssx, ssy])."
+    else
+        ssx = ss[1]; ssy = ss[2]; ssz = ss[3]
+    end
 
     # Get data from .mat file
     path = @__DIR__
     data = MAT.matread(path*"/phantom/brain3D.mat")
-    if us > 1; ss=1; end
-    class = repeat( data["data"][1:ss:end,1:ss:end,start_end[1]:ss:start_end[2]], inner=[us, us, us])
+
+    # subsample or upsample the phantom data
+    class = repeat( data["data"][1:ssx:end,1:ssy:end, start_end[1]:ssz:start_end[2]], inner=[usx, usy, usz])
 
     # Define spin position vectors
-    Δx = .5e-3*ss/us
+    Δx = .5e-3*ssx/usx
+    Δy = .5e-3*ssy/usy
+    Δz = .5e-3*ssz/usz
     M, N, Z = size(class)
     FOVx = (M-1)*Δx #[m]
-    FOVy = (N-1)*Δx #[m]
-	FOVz = (Z-1)*Δx #[m]
+    FOVy = (N-1)*Δy #[m]
+	FOVz = (Z-1)*Δz #[m]
     xx = reshape(-FOVx/2:Δx:FOVx/2,M,1,1) #spin coordinates
-    yy = reshape(-FOVy/2:Δx:FOVy/2,1,N,1) #spin coordinates
-	zz = reshape(-FOVz/2:Δx:FOVz/2,1,1,Z) #spin coordinates
+    yy = reshape(-FOVy/2:Δy:FOVy/2,1,N,1) #spin coordinates
+	zz = reshape(-FOVz/2:Δz:FOVz/2,1,1,Z) #spin coordinates
```

Comment:
There might be some indentation problems here.

Reply:
Yes, I'll make sure I am consistent with the rest of the code going forward. Turning on white space comparison in meld.

Reply:
@cncastillo and @beorostica 

Is there a preference between using tabs or 4-spaces per tab? It looks like a mix of both at the moment at least in Phantom.jl.

Reply:
We are using [JuliaBlue](https://github.com/JuliaDiff/BlueStyle), where the rule is:

> Use 4 spaces per indentation level, no tabs.

My understanding is that Julia's VSCode extension should pick up the [.JuliaFormatter.toml](https://github.com/JuliaHealth/KomaMRI.jl/blob/master/.JuliaFormatter.toml) in the repo, but you can also set it up with a [VSCode config](https://github.com/JuliaDiff/BlueStyle#vs-code-settings).  I am a little surprised that we missed this (#187). We should use https://github.com/julia-actions/julia-format in our workflows.